moj_analytics/auth0_client.py

```diff
@@ -104,26 +104,29 @@ def create(self, resource):
     def get_all(self, resource_class, **kwargs):
         endpoint = '{}s'.format(resource_class.__name__.lower())
 
-        resources = self.request('GET', endpoint, params=kwargs)
+        defaults = {
+            'page': 0,
+            'per_page': 100,
+            'include_totals': True
+        }
 
-        if endpoint in resources:
-            resources = resources[endpoint]
+        params = defaults.copy()
+        params.update(kwargs)
 
-        return [resource_class(self, r) for r in resources]
+        resources = []
 
-    def get_all_pages(self, resource_class, per_page=100, **kwargs):
-        page = 0
         while True:
-            items = self.get_all(
-                resource_class, per_page=per_page, page=page, **kwargs)
+            response = self.request('GET', endpoint, params=params)
 
-            for item in items:
-                yield item
+            if endpoint in response:
+                resources.extend(response[endpoint])
 
-            if len(items) < per_page:
+            if len(resources) == response['total']:
```

Comment:
I think this is fine but *potentially* we could end up with duplicates and missing elements if something is added to a previous page between the two requests...but I don't know if there is any easy way to prevent this and it doesn't sound like something would happen that frequently anyway.

Reply:
Another thing I'm wondering is if this `==` to break out of the loop could potentially cause the loop to never ends if:
1. One of the requests fail and we skip a page (see comment above)
2. An item is added between two requests. If this item would be part of a previous page we may never get it but it would (I'm guessing here) be included/counted by the latest `response['total']` by causing this `==` condition to never be true

I know these are a bit of edge cases.

Reply:
You're right - this code does not handle modification of the list between requests for pages. I will try to make it more robust.